pkg/scheduler/apis/config/validation/validation.go

```diff
@@ -72,6 +78,62 @@ func ValidateKubeSchedulerLeaderElectionConfiguration(cc *config.KubeSchedulerLe
 	return allErrs
 }
 
+func validateTopologySpreadConstraints(constraints []v1.TopologySpreadConstraint, fldPath *field.Path) field.ErrorList {
+	var allErrs field.ErrorList
+	for i, c := range constraints {
+		p := fldPath.Index(i)
+		if c.MaxSkew <= 0 {
+			f := p.Child("maxSkew")
+			allErrs = append(allErrs, field.Invalid(f, c.MaxSkew, "must be greater than zero"))
+		}
+		allErrs = append(allErrs, validateConstraintTopologyKey(c.TopologyKey, p.Child("topologyKey"))...)
+		if err := validateConstraintWhenUnsatisfiable(c.WhenUnsatisfiable, p.Child("whenUnsatisfiable")); err != nil {
+			allErrs = append(allErrs, err)
+		}
+		if c.LabelSelector != nil {
+			f := field.Forbidden(p.Child("labelSelector"), "cluster-level constraint must not define a selector, as they are deduced for each Pod")
+			allErrs = append(allErrs, f)
+		}
+		if err := validateSpreadConstraintNotRepeat(p.Child("{topologyKey, whenUnsatisfiable}"), c, constraints[i+1:]); err != nil {
```

Comment:
make the field path refer to the duplicate constraint index, not a made-up joint key (pass in fldPath and make the error refer to `fldPath.Index(j)` of the duplicate

Reply:
Done